changelogs/unreleased/gh-8958-remove-transaction-nop.md

```diff
@@ -0,0 +1,6 @@
+## bugfix/replication
```

Comment:
Speaking of the second commit. I just had an idea now. Have you considered hacking the transaction flags inside the relay? So it would read the whole txn into memory, find the last global row, and send it with `is_commit`. So no need to store this flag on disk. We simply stop writing NOPs, and the relay will find the last global row itself. Would it work?

Reply:
Good idea, I didn't think of it, actually. Sounds much better than the current approach. Although there will be an added cost of finding the last global row after reading the transaction into memory.

It seems we'll have to read full transactions in relay for the sake of replication filters anyway. To preserve transaction boundaries in filtered streams. So I think it's worth choosing the same approach here. Will try.